hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -73,7 +76,11 @@
      * The default is -1, which means that the value is determined dynamically.
      */
     public static final HazelcastProperty PARTITION_OPERATION_THREAD_COUNT
-            = new HazelcastProperty("hazelcast.operation.thread.count", -1);
+            = new HazelcastProperty("hazelcast.operation.thread.count",
```

Comment:
Protectedion needed against conflicting configurations. So what if the number of threads in the system property doesn't match the number of available cpu's for partition threads.

Reply:
Look at the fail fast checks
https://github.com/hazelcast/hazelcast/pull/16650/files#diff-bad1cef70fcfbb4f7d4ee9b726ed55eeR99

Reply:
If there are checks in place ---> great.